vendors/halo2/src/traits.rs

```diff
@@ -0,0 +1,3 @@
+pub trait RngState {
+    fn state(&self) -> Vec<u8>;
+}
```

Comment:
What about moving this trait to `xor_shift_rng.rs`?

Reply:
it can be shifted there. 
However i want to draw your attenton to a fact that other types except `XORShiftRng` in `xor_shift_rng.rs` might need to implement it, in that case keeping its definition separate would allow implementation independent of  `xor_shift_rng.rs` module.

let me know your view and confirm the changes.

Reply:
Oh I see. But in the case of Halo2, `XorShiftRng` is only used. Could we consider this approach when there should be more implementations to come? Since it looks not natural that `traits.rs` has only this trait even though there are other traits.

Reply:
Sounds cool. l'll move the trait definition to `xor_shift_rng.rs` itself .